JumpscaleCore/servers/gedis/handlers.py

```diff
@@ -300,21 +302,44 @@ def _handle_request(self, request, address, user_session):
         elif request.command.command == "auth":
             tid, seed, signature = request.arguments
             tid = int(tid)
-            try:
-                tclient = j.clients.threebot.client_get(threebot=tid)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            try:
-                verification = tclient.verify_from_threebot(seed, signature)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            # if we get here we know that the user has been authenticated properly
-            user_session.threebot_id = tclient.tid
-            user_session.threebot_name = tclient.name
-            j.shell()
 
+            current_threebot_id = int(j.tools.threebot.me.default.tid)
+
+            # if tid is 0 (special value)
+            # we know that this machine is not registered in phonebook yet
+            # we leave session empty
+            # this is important as some actors may require public access
+            # then it's responsibility of actors to make their own decisions
+            # around who can access them
+            if tid == 0:
```

Comment:
I am ok with this. Please test the following cases then I will approve:
1) using a non-initialized server, get a client to ANOTHER 3bot server. And make sure you can execute any public error.
2) Repeat scenario 1) after initializing the first server, make sure the session is set properly in the second server when executing any actor on it.
3) Test using actors from a local gedis before  setting up the `me`.
4) Test using actors from a local gedis after  setting up the `me`.

@zaibon do you have anything else to add?

Reply:
> @zaibon do you have anything else to add?

LGTM

Reply:
@Hamdy please make sure your tests are done after rebasing from development. I want the tests to be against Andrew's changes too. Thanks!

Reply:
@serboctor @zaibon 
Actualy I found an issue here
If I did something like this locally from kosmos
```
j.tools.threebot.init_my_threebot(myidentity='default', name='hamdy farag', email='ham1dy@d.com', description=None, ipaddr='127.0.0.1', interactive=False)
```
it works, succeeds and I can see that my `j.tools.threebot.me.default.tid` changed
however `3bot Server` will still see ```j.tools.threebot.me.default.tid=0
that breaks stuff, unless `3 Bot server` was restarted!

Reply:
why would 3bots ever work in multiple processes !!!
should never be the case